distribution/lib/Standard/Table/0.0.0-dev/src/Internal/Aggregate_Column_Helper.enso

```diff
@@ -2,71 +2,142 @@ from Standard.Base import all
-    duplicate_names = (if unique.renames.is_empty then [] else [Duplicate_Output_Column_Names unique.renames.to_vector])
-    problems = missing_problems + oob_problems + no_output + invalid_names + duplicate_names
-
-    Validated_Aggregate_Columns key_columns valid_columns problems
+    if renamed_columns.is_empty then
+        problem_builder.report_other_warning No_Output_Columns
+    if unique.invalid_names.not_empty then
+        problem_builder.report_other_warning (Invalid_Output_Column_Names unique.invalid_names.to_vector)
+    if unique.renames.not_empty then
+        problem_builder.report_other_warning (Duplicate_Output_Column_Names unique.renames.to_vector)
+
+    Validated_Aggregate_Columns unique_key_columns renamed_columns problem_builder.build_problemset
+
+## PRIVATE
+   Defines the default name of an `Aggregate_Column`.
+
+   To be used when `new_name` is `Nothing`.
+
+   Assumes that the `Aggregate_Column` is resolved. You may need to transform it
+   with `resolve_columns` first.
+default_aggregate_column_name aggregate_column =
+    case aggregate_column of
+        Group_By c _ -> c.name
+        Count _ -> "Count"
+        Count_Distinct columns _ _ ->
+            "Count Distinct " + (columns.map .name . join " ")
+        Percentile p c _ -> ((p*100).floor.to_text + "%-ile ") + c.name
+        _ ->
+            prefix = Meta.get_simple_type_name aggregate_column . replace "_" " "
+            prefix + " " + aggregate_column.column.name
+
+## PRIVATE
+   Returns a copy of this aggregate where all column descriptors (names,
+   indices or column references potentially from a different table) are
+   replaced with column references from the provided table.
+
+   This preprocess step is required by some helper function, to avoid having
+   to pass the table reference and resolve the column descriptors all the
+   time.
+
+   If some columns cannot be resolved, `Nothing` is returned. Any issues are
+   reported to the `Problem_Builder`.
+resolve_aggregate : Table -> Problem_Builder -> Aggregate_Column -> Aggregate_Column | Nothing
+resolve_aggregate table problem_builder aggregate_column =
+    table_columns = table.columns
+    resolve : (Integer|Text|Column) -> Column ! Internal_Missing_Column_Error
+    resolve c =
+        res = Table_Helpers.resolve_column_helper table_columns c problem_builder
+        if res.is_nothing then Error.throw Internal_Missing_Column_Error else res
+    resolve_selector_to_vector : Column_Selector -> [Column] ! Internal_Missing_Column_Error
+    resolve_selector_to_vector selector =
+        expected_length = case selector of
+            By_Name names _ -> names.length
+            By_Index indexes -> indexes.length
+            By_Column columns -> columns.length
+        resolved = Table_Helpers.select_columns_helper table_columns selector reorder=False problem_builder
+        if resolved.length < expected_length then Error.throw Internal_Missing_Column_Error else resolved
```

Comment:
`>` is possible and should not be covered by this condition.

The `By_Name` matcher can sometimes match _more than one_ column for each matcher, because it can for example have regex matching enabled. And so in this case there are no missing columns. But if less names then expected were matched, it means that some columns were unexpectedly unmatched.

Reply:
Actually discussed with @jdunkerley that we want to proceed even if some columns were not matched (at least in warning mode). So I will update this to `if resolved.is_empty then ...` instead.